app/src/main/res/layout-land/submitted_answer_item.xml

```diff
@@ -25,68 +25,84 @@
+        android:paddingTop="@{viewModel.hasConversationView ? @dimen/padding_16 : @dimen/padding_8}"
+        android:paddingEnd="12dp"
+        android:paddingBottom="@{viewModel.hasConversationView ? @dimen/padding_16 : @dimen/padding_8}"
+        android:text="@{submittedAnswer}"
+        android:textColor="@color/oppiaPrimaryText"
+        android:textSize="16sp"
+        android:visibility="gone"
+        app:layout_constrainedWidth="true"
+        app:layout_constraintHorizontal_bias="0.0"
+        app:layout_constraintEnd_toEndOf="@{viewModel.isCorrectAnswer ? @id/submitted_answer_guideline : ConstraintSet.PARENT_ID}"
+        app:layout_constraintStart_toStartOf="parent"
+        app:layout_constraintTop_toTopOf="parent" />
+
+      <androidx.recyclerview.widget.RecyclerView
+        android:id="@+id/submitted_answer_recycler_view"
+        android:layout_width="match_parent"
+        android:layout_height="wrap_content"
+        android:background="@drawable/submitted_answer_background"
+        android:paddingStart="8dp"
+        android:paddingEnd="8dp"
+        android:visibility="gone"
+        app:itemDecorator="@{@drawable/divider}"
+        app:layoutManager="androidx.recyclerview.widget.LinearLayoutManager"
+        app:layout_constraintStart_toStartOf="parent"
+        app:layout_constraintTop_toTopOf="parent"
+        app:list="@{submittedListAnswer.setOfHtmlStringsList}" />
+
+      <ImageView
+        android:id="@+id/answer_tick"
+        android:layout_width="24dp"
+        android:layout_height="24dp"
+        android:layout_gravity="center_vertical"
+        android:layout_marginStart="12dp"
+        android:src="@drawable/ic_check_primary"
+        android:visibility="@{viewModel.isCorrectAnswer &amp;&amp; !viewModel.hasConversationView ? View.VISIBLE : View.GONE}"
+        app:layout_constraintBottom_toBottomOf="parent"
+        app:layout_constraintStart_toEndOf="@+id/submitted_answer_text_view"
+        app:layout_constraintTop_toTopOf="parent" />
 
-    <ImageView
-      android:id="@+id/answer_tick"
-      android:layout_width="24dp"
-      android:layout_height="24dp"
-      android:layout_gravity="center_vertical"
-      android:layout_marginStart="12dp"
-      android:src="@drawable/ic_check_primary"
-      android:visibility="@{viewModel.isCorrectAnswer &amp;&amp; !viewModel.hasConversationView ? View.VISIBLE : View.GONE}"
-      app:layout_constraintBottom_toBottomOf="parent"
-      app:layout_constraintEnd_toEndOf="parent"
-      app:layout_constraintHorizontal_bias="0.0"
-      app:layout_constraintStart_toEndOf="@+id/submitted_answer_container"
-      app:layout_constraintTop_toTopOf="parent" />
+      <androidx.constraintlayout.widget.Guideline
+        android:id="@+id/submitted_answer_guideline"
+        android:layout_width="wrap_content"
+        android:layout_height="wrap_content"
+        android:orientation="vertical"
+        app:layout_constraintGuide_end="@dimen/margin_36" />
```

Comment:
Shouldn't this be the size of the tick? Also, as I suggested earlier we should be coordinating this with the tick dimensions above. Prefer introducing a new dimension specific to the tick and using that (e.g. ``@dimen/answer_tick_size``).

Reply:
Actually height and width of tick is 24dp, but we also want 12dp padding before it, so that's why I have used 36dp. Should I use 24dp instead? And like 24dp dimension is already defined in dimens.xml, so do I need to create a new dimension with a new name?

Reply:
Ah I missed the margin start of 12dp on the checkbox. This is actually a bit awkward: we have both a width and margin that we need to consider for the guideline. Ideally, we should keep them all in sync with a dimension, but dimensions can't be combined.

Fortunately, data-binding supports adding together dimensions. Unfortunately, it requires a custom binding adapter: https://stackoverflow.com/a/41867789. That might be one way to go. @MohamedMedhat1998 do you have any thoughts on this?

Regarding the dimensions, yes we should add a new, named one. We have a lot of incorrectly named dimensions that lack context (e.g. 'margin_36' doesn't tell you what that margin is actually representing, so when you change it you don't actually know what values are changing). @rt4914 cleaning up our dimension names might be another potential SLoP project.

Reply:
@BenHenning https://github.com/oppia/oppia-android/issues/1646 Does this help? It is not a part of SLoP as the contributor as assigned before that.

Reply:
I think I'm going to complicate things by the following suggestion:  
Why don't we pass the `id` of the view (the tick icon in this case) and let databinding calculate its width and margins/padding instead of manually calculating them (either by using [this approach](https://stackoverflow.com/questions/8101178/how-do-i-apply-mathematical-operations-to-android-dimensions/41867789#41867789), or by creating new value in the xml)?  
If that suggestion isn't applied, I'd stand with the idea of creating a new value in the xml file.

Reply:
I have made some changes here, PTAL

Reply:
I think that would work @MohamedMedhat1998, but it's a bit hard to reason about how databinding's lifecycle interacts with the measure/layout steps of Android views. That being said, I'm not opposed. Maybe take a look at the current solution @prayutsu checked in and see if that's a reasonable substitute to what you're thinking.

@rt4914 we've talked about dimensions a bunch in PRs over the past year, and I think I've actually repeatedly not communicated well what I mean. In this case & in my comment on that issue from a month ago, what I mean by 'specific' dimensions is the dimension should be tied to thing it's representing, *not* its value. For this PR as an example, we should have ``answer_tick_margin`` and ``answer_tick_size`` dimensions that we can reference everywhere that needs to use these sizes to keep consistency. This provides context--it tells you exactly what you're coordinating with. The size itself doesn't matter, and you can always look it up in the dimension file. If it helps, think of these like variable names. They need to have clear context to be useful. Also, consider that dimensions can be dynamically changed for different configurations. If we override '80dp' to mean something else in landscape, that would be really confusing and probably wrong for some of the layouts that use it. Does this help clarify?

@prayutsu please add dimensions as suggested in the previous paragraph & update both the guideline & the answer tick ImageView to use these.

Reply:
> Maybe take a look at the current solution @prayutsu checked in and see if that's a reasonable substitute to what you're thinking.

LGTM



> This provides context--it tells you exactly what you're coordinating with.

I strongly agree with that. We have done that before with the span count and it was really easy to change the span count for different configurations.  
`integers.xml` is a good example of how `dimens.xml` would look like I guess.